utils/fs/memory/memory.go

```diff
@@ -313,6 +291,16 @@ func (c *content) WriteAt(p []byte, off int64) (int, error) {
 	return n, nil
 }
 
+func (c *content) grow(off, l int) {
+	if int(off+l) <= cap(c.bytes) {
+		return
+	}
+
+	buf := make([]byte, 2*len(c.bytes)+l)
```

Comment:
Don't use length and capacity interchangeably.

Even if in this code they always have the same value, they represent quite different things. Choose one and stick to it. The one you are looking for is capacity, not length.